packages/nimble-components/src/table-column/base/tests/table-column.fixtures.ts

```diff
@@ -41,20 +43,66 @@ export class TableColumnEmpty extends TableColumn {
     }
 }
 
-export const tableColumnDelegatesClickAndKeydownTag = 'nimble-test-table-column-delegates';
+const configValidity = ['invalidFoo', 'invalidBar'] as const;
+
 /**
- * Simple empty table column with 'click' and 'keydown' event delegation for testing
+ * Column validator used by TableColumnValidationTest
+ */
+export class TestColumnValidator extends ColumnValidator<
+    typeof configValidity
+> {
+    public constructor(columnInternals: ColumnInternals<unknown>) {
+        super(columnInternals, configValidity);
+    }
+
+    public validateFoo(isValid: boolean): void {
+        this.setConditionValue('invalidFoo', !isValid);
+    }
+
+    public validateBar(isValid: boolean): void {
+        this.setConditionValue('invalidBar', !isValid);
+    }
+}
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-test-table-column-validation': TableColumnValidationTest;
+    }
+}
+
+export const tableColumnValidationTestTag = 'nimble-test-table-column-validation';
+/**
+ * Test column type used to verify column config validation
  */
 @customElement({
-    name: tableColumnDelegatesClickAndKeydownTag
+    name: tableColumnValidationTestTag
 })
-export class TableColumnDelegatesClickAndKeydown extends TableColumn {
+export class TableColumnValidationTest extends TableColumn {
+    @attr({ mode: 'boolean' })
+    public foo: boolean;
+
+    @attr({ mode: 'boolean' })
+    public bar: boolean;
+
+    private readonly validator: TestColumnValidator;
+
     public constructor() {
         super({
             cellRecordFieldNames: [],
             cellViewTag: tableColumnEmptyCellViewTag,
             groupHeaderViewTag: tableColumnEmptyGroupHeaderViewTag,
-            delegatedEvents: ['click', 'keydown']
+            delegatedEvents: []
         });
+        this.validator = new TestColumnValidator(this.columnInternals);
+        this.foo = false;
```

Comment:
Rather than setting `this.foo` and `this.bar` equal to `false` in the constructor, why not assign them the values of `false` where the properties are declared at the top of the class?

Reply:
Because that triggers the `fooChanged` and `barChanged` functions, which then try to use the validator, which is not yet assigned. I considered adding checks in those functions that `this.validator` was set, but I think we want those functions to actually process the initialization of `foo` and `bar` to `false`.

Reply:
I think your idea is right but we shouldn't need the workaround. This doesn't align with our normal attribute assignment pattern.

I think it's the same underlying problem resulting in this workaround in the base class: https://github.com/ni/nimble/blob/4bbe120428b650b5bb528878e21eff7de3edda1b/packages/nimble-components/src/table-column/base/index.ts#L69

Both seem due to the order in which property assignment and constructors run which I wasn't super clear on: https://jsbin.com/gikohuxake/edit?js#

I'm working on a tweak to how column internals is instantiated that I think will address both this cases. More of an FYI, wouldn't block the PR on it but the pattern may change to avoid the workaround shown here.

For the interim can you include a content on these properties being initialized in the constructor to explain the workaround?

Reply:
Added comment